packages/core/src/fst/actor.ts

```diff
@@ -0,0 +1,38 @@
+import { FST } from '.';
+import { EventObject, Observer, SpawnedActorRef } from '..';
+import { toObserver } from '../utils';
+
+export function toActor<TState, TEvent extends EventObject, TOutput = any>(
+  fst: FST<TState, TEvent, TOutput>,
+  handleOutput?: (output: TOutput) => void
+): SpawnedActorRef<TEvent, TState> {
+  let currentState = fst.initialState;
+  const observers = new Set<Observer<TState>>();
```

Comment:
Using Set here is slightly incorrect because each subscription should be properly isolated. In here, when using `subscribe` with **the same** object observer, we won't notify that observer twice when an event is sent to this actor, and also unsubscribing once (for that object observer) will remove that single subscription:
```js
let called = 0
const observer = { next() { called++ } }
const actor = toActor(() => {})
actor.subscribe(observer)
const { unsubscribe } = actor.subscribe(observer)
actor.send({ type: 'FOO' })
expect(called).toBe(2) // actual: 1
unsubscribe()
actor.send({ type: 'FOO' })
expect(called).toBe(3) // actual: 1
```

Reply:
Is there prior art to this? I thought that observers need to be unique.

Reply:
This is how RxJS (among others) handle this sort of a situation: https://codesandbox.io/s/sharp-https-wzk76?file=/src/index.js:191-238

Obervers~ are unique internally so the same external observer creates separate subscriptions. Internally, RxJS converts subscribers to "safe subscribers": https://github.com/ReactiveX/rxjs/blob/5de9e1b355234158ea4320eb4ffd08cbee988c32/src/internal/Observable.ts#L217 and in there all original functions from the observer object are bound https://github.com/ReactiveX/rxjs/blob/5de9e1b355234158ea4320eb4ffd08cbee988c32/src/internal/Subscriber.ts#L168-L170 which returns new references.

To fix that you could just replace this line:
https://github.com/davidkpiano/xstate/blob/a4256dcdbb79a6c19412f01991bff49751d661ca/packages/core/src/utils.ts#L694
with:
```js
    return { next: nextHandler.next, error: nextHandler.error, complete: nextHandler.complete };
```